pkg/services/apiserver/builder/common.go

```diff
@@ -24,13 +23,19 @@ type APIGroupBuilder interface {
 	// Add the kinds to the server scheme
 	InstallSchema(scheme *runtime.Scheme) error
 
-	// Build the group+version behavior
-	GetAPIGroupInfo(
+	// UpdateAPIGroupInfo used to be a getter until we ran into the issue
+	// where separate API Group Info for the same group (different versions) aren't handled well by
+	// the InstallAPIGroup facility of genericapiserver. Also, we can only ever call InstallAPIGroup
+	// once on the genericapiserver per group, or we run into double registration startup errors.
+	//
+	// The caller should share the apiGroupInfo passed into this function across builder versions of the same group.
+	// UpdateAPIGroupInfo builds the group+version behavior updating the passed in apiGroupInfo in place
+	UpdateAPIGroupInfo(
+		apiGroupInfo *genericapiserver.APIGroupInfo,
 		scheme *runtime.Scheme,
-		codecs serializer.CodecFactory,
```

Comment:
`codecs` is not necessary in any iteration of the new usage once apiGroupInfo itself is moved to being created outside the Update func here.